### DISCUSSION TITLE
Self-hosted bit server does not generate previews, documentation, etc...

### DISCUSSION THREAD

**Discussion by AlexThurston** [teambit/bit#8744] [General]
I have a requirement that we be able to run a self-hosted bit server remote to which our components can be exported.  Following the instructions details [here](https://bit.dev/reference/reference/scope/creating-scopes) is straight-forward.  I am able to set that as the remote and when I export, I do see my components listed.

However, there are no details about the component.  Aside from top header information, the rest of the overview page is blank.  

The preview tab shows this and nothing more:
![image](https://github.com/teambit/bit/assets/3537360/066b3e6b-78cc-45a6-97f0-e826e4707324)

Is there a step that I'm missing?

**Comment by nitesh-habilelabs** [comment]
hi @AlexThurston , we also faced same problem a little while ago. to fix this, what we did , is updating the self hosted bit cloud package then tagged and exported those components again with latest version of bit. it fixed the issue. hope this helps you.

**Comment by itaymendel** [comment]
Hi,

preview requries you to build the component. to do it, you need to run `bit tag --build` before you export (if you build from local). otherwise it's better to use the pre-defined CI scripts which already run the build for you.

read more here - https://bit.dev/blog/bit-v10-upgrade-notes-lmq9ec18
